legal-api/src/legal_api/services/compliance/compliance_checks/__init__.py

```diff
@@ -0,0 +1,211 @@
+    NO_PROPRIETOR_ORG_NAME = 'NO_PROPRIETOR_ORG_NAME'
+    NO_PROPRIETOR_ORG_IDENTIFIER = 'NO_PROPRIETOR_ORG_IDENTIFIER'
+    NO_PARTNER = 'NO_PARTNER'
+    NO_PARTNER_PERSON_NAME = 'NO_PARTNER_PERSON_NAME'
+    NO_PARTNER_ORG_NAME = 'NO_PARTNER_ORG_NAME'
+    NO_PARTNER_ORG_IDENTIFIER = 'NO_PARTNER_ORG_IDENTIFIER'
+    NO_COMPLETING_PARTY = 'NO_COMPLETING_PARTY'
+    NO_COMPLETING_PARTY_PERSON_NAME = 'NO_COMPLETING_PARTY_PERSON_NAME'
+    NO_COMPLETING_PARTY_ORG_NAME = 'NO_COMPLETING_PARTY_ORG_NAME'
+    NO_COMPLETING_PARTY_ORG_IDENTIFIER = 'NO_COMPLETING_PARTY_ORG_IDENTIFIER'
+
+    NO_BUSINESS_OFFICE_MAILING_ADDRESS = 'NO_BUSINESS_OFFICE_MAILING_ADDRESS'
+    NO_BUSINESS_OFFICE_MAILING_ADDRESS_STREET = 'NO_BUSINESS_OFFICE_MAILING_ADDRESS_STREET'
+    NO_BUSINESS_OFFICE_MAILING_ADDRESS_CITY = 'NO_BUSINESS_OFFICE_MAILING_ADDRESS_CITY'
+    NO_BUSINESS_OFFICE_MAILING_ADDRESS_COUNTRY = 'NO_BUSINESS_OFFICE_MAILING_ADDRESS_COUNTRY'
+    NO_BUSINESS_OFFICE_MAILING_ADDRESS_POSTAL_CODE = 'NO_BUSINESS_OFFICE_MAILING_ADDRESS_POSTAL_CODE'
+    NO_BUSINESS_OFFICE_MAILING_ADDRESS_REGION = 'NO_BUSINESS_OFFICE_MAILING_ADDRESS_REGION'
+
+    NO_BUSINESS_OFFICE_DELIVERY_ADDRESS = 'NO_BUSINESS_OFFICE_DELIVERY_ADDRESS'
+    NO_BUSINESS_OFFICE_DELIVERY_ADDRESS_STREET = 'NO_BUSINESS_OFFICE_DELIVERY_ADDRESS_STREET'
+    NO_BUSINESS_OFFICE_DELIVERY_ADDRESS_CITY = 'NO_BUSINESS_OFFICE_DELIVERY_ADDRESS_CITY'
+    NO_BUSINESS_OFFICE_DELIVERY_ADDRESS_COUNTRY = 'NO_BUSINESS_OFFICE_DELIVERY_ADDRESS_COUNTRY'
+    NO_BUSINESS_OFFICE_DELIVERY_ADDRESS_POSTAL_CODE = 'NO_BUSINESS_OFFICE_DELIVERY_ADDRESS_POSTAL_CODE'
+    NO_BUSINESS_OFFICE_DELIVERY_ADDRESS_REGION = 'NO_BUSINESS_OFFICE_DELIVERY_ADDRESS_REGION'
+
+    NO_BUSINESS_PARTY_MAILING_ADDRESS = 'NO_BUSINESS_PARTY_MAILING_ADDRESS'
+    NO_BUSINESS_PARTY_MAILING_ADDRESS_STREET = 'NO_BUSINESS_PARTY_MAILING_ADDRESS_STREET'
+    NO_BUSINESS_PARTY_MAILING_ADDRESS_CITY = 'NO_BUSINESS_PARTY_MAILING_ADDRESS_CITY'
+    NO_BUSINESS_PARTY_MAILING_ADDRESS_COUNTRY = 'NO_BUSINESS_PARTY_MAILING_ADDRESS_COUNTRY'
+    NO_BUSINESS_PARTY_MAILING_ADDRESS_POSTAL_CODE = 'NO_BUSINESS_PARTY_MAILING_ADDRESS_POSTAL_CODE'
+    NO_BUSINESS_PARTY_MAILING_ADDRESS_REGION = 'NO_BUSINESS_PARTY_MAILING_ADDRESS_REGION'
+
+    NO_COMPLETING_PARTY_MAILING_ADDRESS = 'NO_COMPLETING_PARTY_MAILING_ADDRESS'
+    NO_COMPLETING_PARTY_MAILING_ADDRESS_STREET = 'NO_COMPLETING_PARTY_MAILING_ADDRESS_STREET'
+    NO_COMPLETING_PARTY_MAILING_ADDRESS_CITY = 'NO_COMPLETING_PARTY_MAILING_ADDRESS_CITY'
+    NO_COMPLETING_PARTY_MAILING_ADDRESS_COUNTRY = 'NO_COMPLETING_PARTY_MAILING_ADDRESS_COUNTRY'
+    NO_COMPLETING_PARTY_MAILING_ADDRESS_POSTAL_CODE = 'NO_COMPLETING_PARTY_MAILING_ADDRESS_POSTAL_CODE'
+    NO_COMPLETING_PARTY_MAILING_ADDRESS_REGION = 'NO_COMPLETING_PARTY_MAILING_ADDRESS_REGION'
+
+
+class ComplianceWarningReferers(str, Enum):
+    """Enum for for compliance warning referers."""
+
+    BUSINESS_OFFICE = 'BUSINESS_OFFICE'
+    BUSINESS_PARTY = 'BUSINESS_PARTY'
```

Comment:
What is Business Party? A Proprietor or Partner? (Would this be a better enum? Or is business party a common business term?)

Reply:
Could this include directors in the future?

And could address be Registered Office or Records Office in the future?

Reply:
> What is Business Party? A Proprietor or Partner? (Would this be a better enum? Or is business party a common business term?)

`ComplianceWarningReferers` is what I came up with as an attempt to make it easier to deliver specific error codes and warning messages without creating a mess.  It was getting particular messy with the different types of messages.  

Business Party is intended to reference essentially entries in the `party_roles` table.  For firms this would be partners and proprietors.  It could be directors too.  But yes, if we wanted to get really granular I could break down `BUSINESS_PARTY` down to `PARTNER` and `PROPRIETOR` and get more specific with these compliancy warnings.  I only really did it for addresses for now.
![image](https://user-images.githubusercontent.com/6685223/168381605-8020595e-1830-49ea-9be7-0243cdcb4d64.png)

Reply:
> Could this include directors in the future?

Yes, as per my previous comment `BUSINESS_PARTY` could be broken down to include `DIRECTORS`.

> And could address be Registered Office or Records Office in the future?
And yes, we could add `REGISTERED_OFFICE` and `RECORDS_OFFICE` to `ComplianceWarningReferers` and update  `REFERER_WARNINGS_MAPPING ` in the future.

Reply:
Sounds good. Looks OK for now!